src/pds_doi_service/core/outputs/doi_validator.py

```diff
@@ -293,84 +295,92 @@ def _check_lidvid_field(self, doi: Doi):
+                    "Version identifier must consist only of a major and minor version "
+                    "joined with a period (ex: 1.0)"
                 )
         except InvalidIdentifierException as err:
             raise InvalidIdentifierException(
-                f'The record identifier {doi.related_identifier} (DOI {doi.doi}) '
-                f'does not conform to a valid LIDVID format.\n'
-                f'Reason: {str(err)}\n'
-                'If the identifier is not intended to be a LIDVID, use the '
-                'force option to bypass the results of this check.'
+                f"The record identifier {doi.related_identifier} (DOI {doi.doi}) "
+                f"does not conform to a valid LIDVID format.\n"
+                f"Reason: {str(err)}\n"
+                "If the identifier is not intended to be a LIDVID, use the "
+                "force option to bypass the results of this check."
             )
 
     def _check_field_workflow(self, doi: Doi):
         """
         Check that there is not a record in the sqllite database with same
         identifier but a higher status than the current action (see workflow_order)
         """
-        if doi.status.lower() not in self.m_workflow_order:
-            msg = (f"Unexpected DOI status of '{doi.status.lower()}' from label. "
-                   f"Valid values are "
-                   f"{[DoiStatus(key).value for key in self.m_workflow_order.keys()]}")
+        if doi.status is not None and doi.status.lower() not in self.m_workflow_order:
+            msg = (
+                f"Unexpected DOI status of '{doi.status.lower()}' from label. "
+                f"Valid values are "
+                f"{[DoiStatus(key).value for key in self.m_workflow_order.keys()]}"
+            )
             logger.error(msg)
             raise UnexpectedDOIActionException(msg)
 
         # The database expects each field to be a list.
-        query_criterias = {'ids': [doi.related_identifier]}
+        query_criterias = {"ids": [doi.related_identifier]}
 
         # Query database for rows with given id value.
         columns, rows = self._database_obj.select_latest_rows(query_criterias)
 
         if rows:
             row = rows[0]
-            doi_str = row[columns.index('doi')]
-            prev_status = row[columns.index('status')]
+            doi_str = row[columns.index("doi")]
+            prev_status = row[columns.index("status")]
 
             # A status tuple of ('Pending',3) is higher than ('Draft',2) will
             # cause an error.
-            if self.m_workflow_order[prev_status.lower()] > self.m_workflow_order[doi.status.lower()]:
+            #
+            # 🤔 TODO: ``mypy`` has several complaints about this line:
+            # • doi.status is an optional (``None``) so calling ``lower`` on it could fail; there should be a check
+            # • The indexing on ``DoiStatus`` here is by ``str``, but is declared to be ``DoiStatus``
```

Comment:
Interesting find, we (or I) could address the first bullet by making `DoiStatus.Unknown` the default value for `status` in the `Doi` dataclass. Next, I can at least speak to the second bullet, in that `DoiStatus` is an inheritor of both `str` and `Enum`, which is why it should be safe to call `lower()`, as long as the value isn't `None`.